src/fsharp/ast.fs

```diff
@@ -1365,18 +1366,18 @@ and SynModuleDecls = SynModuleDecl list
 and 
     [<NoEquality; NoComparison>]
     SynExceptionSig = 
-    | SynExceptionSig of SynExceptionDefnRepr * SynMemberSigs * range
+    | SynExceptionSig of SynExceptionDefnRepr * SynMemberSigs * range:range
 
 and
     [<NoEquality; NoComparison; RequireQualifiedAccess>]
     SynModuleSigDecl =
-    | ModuleAbbrev      of Ident * LongIdent * range
-    | NestedModule      of SynComponentInfo * isRec : bool * SynModuleSigDecls * range
-    | Val               of SynValSig * range
-    | Types             of SynTypeDefnSig list * range
-    | Exception         of SynExceptionSig * range
-    | Open              of LongIdent * range
-    | HashDirective     of ParsedHashDirective * range
+    | ModuleAbbrev      of ident:Ident * longId:LongIdent * range:range
+    | NestedModule      of SynComponentInfo * isRecursive:bool * SynModuleSigDecls * range:range
+    | Val               of SynValSig * range:range
```

Comment:
Yes in theory but we normally rely on Git to mark up conflicts when we do merges

Note we normally merge visualfsharp --> fsharp --> FCS. 

Because we already allowed these additions to FCS then we will get conflicts when we merge a conflicting version of these changes into visualfsharp.

Basically every time we change FCS without first submitting to visualfsharp we build up lots of work for ourselves.  It's useful from time to time (especially when engineering in this repo was hard and CI incomplete) but we pay a price for it.